Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Azure.CosmosRepository.Extensions;
+
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <inheritdoc/>
+    public class InMemoryRepository<TItem> : IRepository<TItem> where TItem : IItem
+    {
+        internal List<TItem> Items { get; } = new();
```

Comment:
One consideration, you might want to use a `ConcurrentDictionary<string, TItem>` instead. Where the key is the `IItem.Id`. This would help ensure that these are thread-safe, and since it is just for testing it's not a major perf concern to have the extra overhead of that data structure.

Reply:
I did think about this, I thought due to the use of the in-memory option being used for more test/demo that thread contention may not be an issue, how would a `ConcurrentDictionary<string, TItem>` behave with the Expressions was my main concern. I may be wrong but was not sure you can do Linq on these can you? I will double check.

Reply:
It can be done the same way that you've done it now, with the Compile call to the expressions. Then it's just linq over the concurrent dictionary.

Reply:
Okay that sounds good, will make this change, get them done by the evening, thanks for your feedback.

Reply:
This has been implemented now @IEvangelist